src/api/ucc.h

```diff
@@ -0,0 +1,1497 @@
+ *  @param [in]  print_flags   Flags that control various printing options.
+ *
+ *  @parblock
+ *
+ *  @b Description
+ *
+ *  The routine prints the configuration information that is stored in
+ *  ucc_lib_config_h "configuration" descriptor.
+ *
+ *  @endparblock
+ *
+ */
+
+void ucc_lib_config_print(const ucc_lib_config_h config, FILE *stream,
+                          const char *title, ucc_config_print_flags_t print_flags);
+
+/**
+ *  @ingroup UCC_LIB
+ *
+ *  @brief The @ref ucc_lib_config_modify routine modifies the runtime configuration as described by the
+ *  descriptor.
+ *
+ *  @param [in] config   Pointer to the configuration descriptor to be modified
+ *  @param [in] name     Configuration variable to be modified
+ *  @param [in] value    Configuration value to set
+ *
+ *  @parblock
+ *
+ *  @b Description
+ *
+ *  The @ref ucc_lib_config_modify routine sets the value of identifier "name"
+ *  to "value".
+ *
+ *  @endparblock
+ *
+ *  @return Error code as defined by ucc_status_t
+ */
+
+ucc_status_t ucc_lib_config_modify(ucc_lib_config_h *config, const char *name,
+                                   const char *value);
+
+
+/**
+ *  @ingroup UCC_LIB
+ *
+ *  @brief The @ref ucc_init initializes the UCC library.
+ *
+ *
+ *  @param [in]  params    user provided parameters to customize the library functionality
+ *  @param [in]  config    UCC configuration descriptor allocated through
+ *                         @ref ucc_lib_config_read "ucc_config_read()" routine.
+ *  @param [out] lib_p     UCC library handle
+ *
+ *  @parblock
+ *
+ *  @b Description
+ *
+ *  A local operation to initialize and allocate the resources for the UCC
+ *  operations. The parameters passed using the ucc_lib_params_t and
+ *  @ref ucc_lib_config_h structures will customize and select the functionality of the
+ *  UCC library. The library can be customized for its interaction with the user
+ *  threads, types of collective operations, and reductions supported.
+ *  On success, the library object will be created and ucc_status_t will return
+ *  UCC_OK. On error, the library object will not be created and corresponding
+ *  error code as defined by ucc_status_t is returned.
+ *
+ *  @endparblock
+ *
+ *  @return Error code as defined by ucc_status_t
+ *
+ */
+
+ucc_status_t ucc_init(const ucc_lib_params_t *params,
+                      const ucc_lib_config_h *config,
+                      ucc_lib_h *lib_p);
+
+/**
+ *  @ingroup UCC_LIB
+ *
+ *  @brief The @ref ucc_finalize routine finalizes the UCC library.
+ *
+ *  @param [in] lib_p   Handle to ucc_lib_h
+ *                      "UCC library".
+ *
+ *  @parblock
+ *
+ *  @b Description
+ *
+ *  A local operation to release the resources and
+ *  cleanup. All participants that invoked @ref ucc_init should call this
+ *  routine.
+ *
+ *  @endparblock
+ *
+ *  @return Error code as defined by ucc_status_t
+ */
+
+ucc_status_t ucc_finalize(ucc_lib_h lib_p);
```

Comment:
I don't see any reason for destroy/finalize functions to return error code except for debugging. So alternatively to status return these functions might print kind of warning in case something went wrong.

Reply:
What is advantage of returning void ? If you don't need it, you can always ignore it. :) I'm missing something.